trunk/src/protocol/srs_protocol_rtmp_stack.cpp

```diff
@@ -1137,11 +1138,18 @@ srs_error_t SrsProtocol::read_message_header(SrsChunkStream* chunk, char fmt)
          * about the is_first_chunk_of_msg.
          * @remark, for the first chunk of message, always use the extended timestamp.
          */
-        if (!is_first_chunk_of_msg && chunk_timestamp > 0 && chunk_timestamp != timestamp) {
+        if (!is_first_chunk_of_msg && chunk_extended_timestamp > 0 && chunk_extended_timestamp != timestamp) {
+        // if (!is_first_chunk_of_msg && chunk_timestamp > 0 && chunk_timestamp != timestamp) {
             mh_size -= 4;
             in_buffer->skip(-4);
         } else {
-            chunk->header.timestamp = timestamp;
+            chunk->extended_timestamp = timestamp;
+            if (fmt == RTMP_FMT_TYPE0) {
```

Comment:
There is still a coding style issue, please just copy the below lines of code:

```cpp

            if (fmt == RTMP_FMT_TYPE0) {
                chunk->header.timestamp = timestamp;
            } else if (is_first_chunk_of_msg) {
                chunk->header.timestamp += timestamp;
            }
```

Reply:
Thank you very much. We are also introducing the clang-format tool into our workflow, thanks to your suggestion.